Lib/asyncio/stack.py

```diff
@@ -0,0 +1,231 @@
+"""Introspection utils for tasks call stacks."""
+
+import sys
+import types
+import typing
+
+from . import events
+from . import futures
+from . import tasks
+
+__all__ = (
+    'capture_call_graph',
+    'print_call_graph',
+    'FrameCallGraphEntry',
+    'FutureCallGraph',
+)
+
+# Sadly, we can't re-use the traceback's module datastructures as those
+# are tailored for error reporting, whereas we need to represent an
+# async call stack.
+#
+# Going with pretty verbose names as we'd like to export them to the
+# top level asyncio namespace, and want to avoid future name clashes.
+
+
+class FrameCallGraphEntry(typing.NamedTuple):
```

Comment:
Why not a dataclass? namedtuple can break when unpacked as tuple if in future a new attribute is added

Reply:
Hm, this is a valid argument. I don't expect people to use these as tuples, but undoubtedly some will. Please take a look at my above comment on named tuples, if you still feel strongly about it I can change this to dataclasses. Good point.

Reply:
One datapoint is that dataclasses will be slighly smaller:

```
>>> @dataclasses.dataclass
... class A:
...    x: object
...    y: object
...    z: object
...
>>> A2 = collections.namedtuple("B", 'x y z')
>>> import sys
>>> sys.getsizeof(A2(1,2,3))
64
>>> sys.getsizeof(A(1,2,3))
48
```

Reply:
OTOH importing dataclasses will slow down importing this

Reply:
@pablogsal I think @kumaraditya303 is right here and we should use dataclasses. He's right in that some people *will* use tuple unpacking and that somewhat closes the door for us to evolve this API. I'll push an update.

Reply:
There is bug in the dataclass size. The size must be at least 72 and is probably 80.
As well as the a,b,c fields there are the weakref, dict, two GC fields, refcount and class fields.

It shouldn't really impact your decision either way, but if size matters it slightly favors namedtuples.
If size really matters, use `__slot__`s.